### DISCUSSION TITLE
Spatial aggregation

### DISCUSSION THREAD

**Discussion by ecomodeller** [DHI/modelskill#12] [Ideas]
In case one has a lot of measurements, either from many stations or from satellites, it is useful to aggregate the model skill in defined subdomains.
![image](https://user-images.githubusercontent.com/614215/107620971-d6669880-6c55-11eb-9066-17df7b350563.png)

**Comment by Hendrik1987** [comment]
For satellite data, this is exactly what we have done for the GWM, building on existing MATLAB tools. One way of aggregation were regular lon/lat bins with results like this example. Having an improved support for altimetry data would be great and something I could happily contribute to.
![GWM png](https://user-images.githubusercontent.com/29595985/108681817-e4d37080-74ef-11eb-9502-9b7887dfb3fa.jpg)

**Comment by jsmariegaard** [comment]
I guess this is "solved" now by the area argument to the different skill and scatter methods (filtering) and by spatial_skill() from #38.

**Reply by Hendrik1987** [reply]
I think yes and no.

Yes, a skill dataframe per subdomain can already be achieved in 2 ways:

1. Loop over a list of areas and stack the resulting skill dataframes.
2. Extact the data with .all_df(), add a subdomain category column and add the column to the `by` argument of .skill() (similar to the low and high sea state example).

However, I think a more user friendly solution would be nice; And maybe a method to create a plot similar to above example. 
The xarray based `spatial_skill()` is only suitable for a spatial aggregation on a regular grid.

One option could be to allow a list of areas as `area` argument. `skill()` would return a dataframe with `subdomain` as additional multi-index, `scatter()` could produce one scatter plot per domain (optionally as subplots), `hist()` could have stacked histograms and `spatial_skill()` could return a list of datasets.

I like the simplicity of this solution. I don't like that it would mix the selection and grouping part, which could be confusing. It is also very specific to the subdomain use case. Grouping by other variables or attributes would still need other solutions.

So, maybe it's time to think about a smart, scalable way to add categories of various types.
